lib/ohai/plugins/darwin/hardware.rb

```diff
@@ -0,0 +1,93 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2016-present Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Hardware) do
+  provides 'hardware'
+
+  collect_data(:darwin) do
+    begin
+      require "plist"
+    rescue Gem::LoadError
+      # In case the plist gem isn't present, skip this plugin.
+      next
+    end
+
+    hardware Mash.new
```

Comment:
hardware is a good high level namespace. The subkeys seem to be pretty clean here. If someone later on wanted to add something like HP iLO on the Linux side there'd be room and it wouldn't be horribly cluttered.